ee/clickhouse/queries/retention/retention_event_query.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Dict, Literal, Tuple
+
+from ee.clickhouse.models.action import format_action_filter
+from ee.clickhouse.queries.event_query import ClickhouseEventQuery
+from ee.clickhouse.queries.util import get_trunc_func_ch
+from posthog.constants import PAGEVIEW_EVENT, TREND_FILTER_TYPE_ACTIONS, TREND_FILTER_TYPE_EVENTS, TRENDS_LINEAR
+from posthog.models import Entity
+from posthog.models.action import Action
+from posthog.models.filters.retention_filter import RetentionFilter
+
+
+class RetentionEventsQuery(ClickhouseEventQuery):
+    _filter: RetentionFilter
+    _event_query_type: Literal["returning", "target", "target_first_time"]
```

Comment:
Make this an enum? Makes using it in retention class much easier as well, and improves typing 😅